src/cargo/core/compiler/mod.rs

```diff
@@ -649,6 +667,13 @@ fn add_plugin_deps(
     Ok(())
 }
 
+fn get_dynamic_search_path(path: &PathBuf) -> &Path {
```

Comment:
Would you mind moving this function extraction to its own commit? That would help review and trace code in the future.

(Git is pretty bad at displaying this kind of diff 😞).